src/Autocomplete/AutocompleteInput.tsx

```diff
@@ -56,13 +58,13 @@ const AutocompleteInput = React.forwardRef(
         // HACK: wait a tick and check the focused element before hiding the autocomplete menu
         // this prevents the menu from hiding when the user is clicking an option in the Autoselect.Menu,
         // but still hides the menu when the user blurs the input by tabbing out or clicking somewhere else on the page
-        setTimeout(() => {
+        safeSetTimeout(() => {
```

Comment:
It would be even better if we could refactor this setTimeout away entirely. `InlineAutocomplete` managed to do it with a synthetic change event: https://github.com/primer/react/blob/main/src/drafts/InlineAutocomplete/InlineAutocomplete.tsx#L155

I don't think that should block the rest of these changes though.